### ISSUE TITLE
SASS files get cached even though SASS compiler is not included in COMPRESS_CACHEABLE_PRECOMPILERS

### ISSUE THREAD

**Issue by Senthe** [django-compressor/django-compressor#772]
My SASS file is an usual `app.sass` which contains only `@include`s for partials. When I set `COMPRESS_ENABLED = True` and change one of the partials, file doesn't get recompiled and I am served a cached file instead. This is specifically discussed by docs which state that the file should be recompiled on every request as long as compiler is not included in `COMPRESS_CACHEABLE_PRECOMPILERS`. Well this is not true in my case. I have to manually delete CSS file from Redis every time I want to have it recompiled.

When I don't set `COMPRESS_ENABLED = True` everything works fine (SASS is recompiled every time), but my JS files are not compressed into one file anymore, so I have 50 requests instead of one which makes page load 10 times slower. This is a pain during development.

My setup:
```
COMPRESS_PRECOMPILERS = (
    ('text/coffeescript', 'coffee --compile --stdio'),
    ('text/sass', 'django_libsass.SassCompiler'),
)
COMPRESS_CACHE_BACKEND = 'compressor'
COMPRESS_CACHEABLE_PRECOMPILERS = (
    'text/coffeescript',
)
COMPRESS_ENABLED = True
```

My Jade file:
html
  head
    {% compress css %}
    link(rel='stylesheet', href='/static/main/stylesheets/app.sass', type='text/sass', charset='utf-8')
    {% endcompress %}
    {% compress js %}
    ...
    {% endcompress %}

**Comment by karyon** [comment]
i just checked and didnt see anything obvious. im pretty sure that the `COMPRESS_CACHEABLE_PRECOMPILERS` is handled correctly, see https://github.com/django-compressor/django-compressor/blob/f5c1469c8ebca2ad443066bd283cecd9b57ae3df/compressor/filters/base.py#L220 clearly the cache is not used there if the setting is disabled, and in your case it shouldnt even get there as that seems to apply only for precompilers that use a command, not a python module.

so yeah, no idea. this needs some debugging and i likely wont have the time to do that anytime soon. if you could have a look at this by yourself that would be awesome!

**Comment by karyon** [comment]
@senthe, @lemuelf, @shirazz, @Remiz, has any of you found anything that might help debug this?

**Comment by Senthe** [comment]
As for me, I am not working on this project anymore and I didn't find anything in the time I was.

**Comment by Remiz** [comment]
I only use compress on production and to get over that issue I added a command to perform a touch on every less/sass files in my project directory before deploying (I use fabric for deployment). 
I feel everything that is referenced via @include is not recognized at all by compressor, only the files referenced within the compress tags are.

**Comment by karyon** [comment]
@Remiz that is correct, django-compressor doesn't know of any include mechanisms inside js/css/coffescript/less/whatever files.

as i understood this issue, the problem is that, if the precompiler is not in the cacheable setting, these files should be compiled on every request *regardless* whether they have changed or not, but they are not recompiled at all. have you experienced that as well or is your problem the includes?

**Comment by WeirdCarrotMonster** [comment]
The culprit is `Compressor.cachekey` property. As neither filename nor mtime of root sass file changes, `cache_key` in `CompressorMixin.render_compressed` is loaded from cache. As a workaround for development `DummyCache` backend can be used to avoid caching keys entirely.
